model/src/main/kotlin/VcsType.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Copyright (C) 2019 Bosch Software Innovations GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.model
+
+import com.fasterxml.jackson.annotation.JsonCreator
+import com.fasterxml.jackson.annotation.JsonValue
+
+/**
+ * An class for Version Control System types. Each type has one or more [aliases] associated to it, where the first
+ * alias is the definite name.
+ */
+data class VcsType(val aliases: List<String>) {
+    @JsonCreator
+    constructor(type: String): this(
+        ALL_ALIASES.find { aliases ->
+            aliases.find { alias ->
+                alias.equals(type, true)
```

Comment:
- Using named parameters for the equals call would imo increase readability.
- Shall we really completely ignore the case? 
- I'd prefer specifying the exact strings in the list to be a bit more 
strict.
- Is it needed to use aliases at all? If not I'd vote for not adding them at all.

Reply:
> Using named parameters for the equals call would imo increase readability.

We currently handle that inconsistently throughout the code base, but I agree that we should probably have a convention to always use named arguments for booleans, so I'll change it.

> Shall we really completely ignore the case?

Yes. That's the main point of this change: To avoid the confusion whether you need to write `type == "git"` or `type == "Git"` if you want to check for the VCS type.

> I'd prefer specifying the exact strings in the list to be a bit more
strict.

You mean to explicitly spell out all possible capitalizations, incl. "giT" etc.? I'm against that as that would make the list explode.

> Is it needed to use aliases at all? If not I'd vote for not adding them at all.

Yes. Some package managers use e.g. "Subversion" in their meta-data to describe the VCS, some use "SVN". We need a way to normalize these descriptions to a single type so we know how to clone the source, and for that we need the aliases.